api/integrations/github/client.py

```diff
@@ -12,3 +54,154 @@ def generate_token(installation_id: str, app_id: int) -> str:
+    headers = build_request_headers(installation_id, use_jwt=True)
+    response = requests.delete(url, headers=headers, timeout=GITHUB_API_CALLS_TIMEOUT)
+    response.raise_for_status()
+    return response
+
+
+def fetch_github_resource(
+    resource_type: ResourceType,
+    organisation_id: int,
+    params: RepoQueryParams,
+) -> Response:
+    github_configuration = GithubConfiguration.objects.get(
+        organisation_id=organisation_id, deleted_at__isnull=True
+    )
+    # Build Github search query
+    q = ["q="]
+    if params.search_text:
+        q.append(params.search_text)
+    q.append(f"repo:{params.repo_owner}/{params.repo_name}")
+    q.append(f"is:{resource_type.value}")
+    if params.state:
+        q.append(f"is:{params.state}")
+    q.append("in:title")
+    if params.search_in_body:
+        q.append("in:body")
+    if params.search_in_comments:
+        q.append("in:comments")
+    if params.author:
+        q.append(f"author:{params.author}")
+    if params.assignee:
+        q.append(f"assignee:{params.assignee}")
+
+    url = (
+        f"{GITHUB_API_URL}search/issues?"
+        + " ".join(q)
+        + f"&per_page={params.page_size}&page={params.page}"
+    )
+    headers: dict[str, str] = build_request_headers(
+        github_configuration.installation_id
+    )
+    response = requests.get(url, headers=headers, timeout=GITHUB_API_CALLS_TIMEOUT)
+    response.raise_for_status()
+    json_response = response.json()
+    results = [
+        {
+            "html_url": i["html_url"],
+            "id": i["id"],
+            "title": i["title"],
+            "number": i["number"],
+        }
+        for i in json_response["items"]
+    ]
+    data = {
+        "results": results,
+        "count": json_response["total_count"],
+        "incomplete_results": json_response["incomplete_results"],
+    }
+    if response.links.get("prev"):
+        data["previous"] = response.links.get("prev")
+
+    if response.links.get("next"):
+        data["next"] = response.links.get("next")
+
+    # Return a Response object
+    return Response(
+        data=data,
+        content_type="application/json",
+        status=status.HTTP_200_OK,
+    )
+
+
+def fetch_github_repositories(installation_id: str) -> Response:
+    url = f"{GITHUB_API_URL}installation/repositories"
+
+    headers: dict[str, str] = build_request_headers(installation_id)
+
+    response = requests.get(url, headers=headers, timeout=GITHUB_API_CALLS_TIMEOUT)
+    json_response = response.json()
+    response.raise_for_status()
+    results = [
+        {
+            "full_name": i["full_name"],
+            "id": i["id"],
+            "name": i["name"],
+        }
+        for i in json_response["repositories"]
+    ]
+    data = {
+        "repositories": results,
+        "total_count": json_response["total_count"],
+    }
+    return Response(
+        data=data,
+        content_type="application/json",
+        status=status.HTTP_200_OK,
+    )
```

Comment:
Same as [this comment](https://github.com/Flagsmith/flagsmith/pull/3962/files#r1609687693).

Reply:
ok, Done